### PR TITLE
[MODEXPW-94] - Upgraded spring-boot-starter-parent from 2.5.2 to 2.6.6

### PR DESCRIPTION
**Pull request by gurleenkaurbp**
[MODEXPW-94] - Upgraded spring-boot-starter-parent from 2.5.2 to 2.6.7 and also updated the folio-spring-base.version to 3.0.3.
(https://issues.folio.org/browse/MODEXPW-94)

## Purpose
The spring-boot-starter-parent has to be upgraded from 2.5.2 to 2.6.7 to overcome the vulnerability of Spring4Shell in mod-data-export-worker and also updated the folio-spring-base.version to 3.0.3.

## Approach
Upgraded spring-boot-starter-parent from 2.5.2 to 2.6.7 and also updated the folio-spring-base.version to 3.0.3. in pom.xml

### CODE REVIEW

**Review by siarhei-charniak — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
-        <version>2.6.2</version>
```
Comment (alekGbuz) [inline pom.xml:8]: Used folio-spring-base library of 3.0.1 has spring-boot-starter-parent 2.6.2.  folio-spring-base  version should be upgraded also.

File: pom.xml [hunk 2]
```diff
@@ -5,7 +5,7 @@
...
+        <version>2.6.6</version>
```
Comment (alekGbuz) [inline pom.xml:8]: Please review possibility of usage 2.6.7 version of spring-boot-starter-parent. cc @julianladisch

File: pom.xml (hunk 1 above)
Comment (gurleenkaurbp) [reply pom.xml:8]: Changes made to use the latest version of folio-spring-base.version 3.0.3. File committed for the same.

File: pom.xml (hunk 2 above)
Comment (gurleenkaurbp) [reply pom.xml]: Yes, made the version to 2.6.7 as folio-spring-base.version 3.0.3 has the version 2.6.7 changes were made as part of FOLSPRINGB-51

**Pull request merged**